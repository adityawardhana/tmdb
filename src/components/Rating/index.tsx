import React, { FC } from "react";
import { IoIosStar } from "react-icons/io";

interface Props {
  rating?: number;
  review?: number;
}

const Rating: FC<Props> = ({ rating, review }) => {
  if (!rating) {
    return <div />;
  }
  return (
    <div className="flex items-center gap-2" data-testid="rating">
      <div className="flex items-start">
        <IoIosStar className="w-4 h-4 text-yellow-300 me-1" />
        <p className="ms-2 text-sm font-bold text-gray-900 dark:text-white">
          {rating.toFixed(2)}
        </p>
      </div>
      {review && (
        <div className="text-xs text-gray-900 underline hover:no-underline dark:text-white">
          ({review.toLocaleString()} reviews)
        </div>
      )}
    </div>
  );
};

export default Rating;
