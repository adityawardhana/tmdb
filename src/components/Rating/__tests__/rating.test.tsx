import { render, screen } from "@testing-library/react";
import Rating from "..";

it("render Rating not found", () => {
  const { container } = render(<Rating />);

  expect(container).toMatchSnapshot();
});
