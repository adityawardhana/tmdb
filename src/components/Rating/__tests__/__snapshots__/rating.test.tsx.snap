// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Rating not found 1`] = `
<div>
  <div />
</div>
`;
